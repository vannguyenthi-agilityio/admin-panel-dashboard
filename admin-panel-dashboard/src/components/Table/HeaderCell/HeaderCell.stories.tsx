// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import HeaderCell from ".";

const meta: Meta<typeof HeaderCell> = {
  title: "Components/Table/HeaderCell",
  tags: ["autodocs"],
  component: HeaderCell,
  argTypes: {
    title: {
      description: "Title of header cell content",
    },
    isSortable: {
      description: "Have or not isSortable by set it as true or false",
    },
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

export const HeaderCellHaveSortIncrease: Story = {
  args: {
    title: "Customer",
    isSortable: true,
  },
};
export const HeaderCellHaveNoSort: Story = {
  args: {
    title: "Customer",
    isSortable: false,
  },
};
