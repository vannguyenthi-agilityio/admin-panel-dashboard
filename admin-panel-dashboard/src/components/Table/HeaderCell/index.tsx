import { memo } from "react";

// Components
import { Button } from "@/components";

// Icons
import { SortIcon } from "@/components/Icons";

export interface IHeaderCell {
  title: string;
  isSortable: boolean;
}

const HeaderCell = ({
  title,
  isSortable,
}: IHeaderCell) => {
  return (
    <div className="flex items-center w-full">
      <div className="text-sm font-normal text-darker">
        {title}
      </div>
      {isSortable && (
        <Button
          buttonType='transprent'
          className='w-[16px] h-[16px] p-0 ml-2'
          aria-label="Arrow Sort"
          leftIcon={<SortIcon />}
        />
      )}
    </div>
  );
};

export default memo(HeaderCell);
