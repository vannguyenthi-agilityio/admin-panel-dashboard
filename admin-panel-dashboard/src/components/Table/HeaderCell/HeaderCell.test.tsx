import { render } from "@testing-library/react";

// Components
import HeaderCell from ".";

describe("HeaderCell component", () => {
  const propsHaveSortHeaderCell = {
    title: "Customer",
    isSortable: true,
  };

  it("should render snapshot correctly", () => {
    const { container } = render(
      <HeaderCell {...propsHaveSortHeaderCell} />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should have Sort button if isSortable is true", () => {
    const { getByLabelText } = render(
      <HeaderCell {...propsHaveSortHeaderCell} />,
    );

    expect(getByLabelText("Arrow Sort")).toBeTruthy();
  });

  it("should not have Sort button if isSortable is false", () => {
    const propsIncreaseTypeHeaderCell = {
      title: "",
      isSortable: false,
    };

    const { queryByLabelText } = render(
      <HeaderCell {...propsIncreaseTypeHeaderCell} />,
    );

    expect(queryByLabelText("Arrow Sort")).toBeFalsy();
  });
});
