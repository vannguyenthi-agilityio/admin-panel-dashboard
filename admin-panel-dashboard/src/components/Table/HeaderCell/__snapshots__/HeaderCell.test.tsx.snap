// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCell component should render snapshot correctly 1`] = `
<div>
  <div
    class="flex items-center w-full"
  >
    <div
      class="text-sm font-normal text-darker"
    >
      Customer
    </div>
    <button
      aria-label="Arrow Sort"
      class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-transprent border-none hover:bg-transprent text-sm rounded-lg w-[16px] h-[16px] p-0 ml-2"
      type="button"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.66669 5.16666H13.3334M4.33335 7.83332H11.6667M5.66669 10.5H10.3334"
          stroke="#6c757d"
          stroke-linecap="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
